llvm/test/CodeGen/AMDGPU/llc-pipeline.ll

```diff
@@ -362,10 +362,12 @@
 ; GCN-O1-NEXT:        Machine Optimization Remark Emitter
 ; GCN-O1-NEXT:        Greedy Register Allocator
 ; GCN-O1-NEXT:        Virtual Register Rewriter
+; GCN-O1-NEXT:        Stack Slot Coloring
 ; GCN-O1-NEXT:        SI lower SGPR spill instructions
 ; GCN-O1-NEXT:        Virtual Register Map
 ; GCN-O1-NEXT:        Live Register Matrix
 ; GCN-O1-NEXT:        SI Pre-allocate WWM Registers
+; GCN-O1-NEXT:        Live Stack Slot Analysis
```

Comment:
Can we keep this preserved in the previous passes?

Reply:
I tried this & found that as StackSlotColoring pass is not currently updating LiveStacks(LS) analysis, eventhough modifying the stack slots usage (as before it is being invoked only after complete RA when all spills happened just before PEI, so LS not needed anymore), so LS results are invalidated rather preserved. But, now LS is being needed by RA (for remaining register class) later, so LS analysis appears later as being re-calculated.

Reply:
Right, can the pass be updated so that it can be preserved?

Reply:
Need to look into it, but I cannot say how much changes might it take. Also, as the same LiveStack(LS) would be propagated into RA pass later for entring additional entries, RA might have some side effects while working on presereved LS. Considering all this, this updates might take big significant effort, followed by rigourous tests, so nothing else breaks.

Reply:
@arsenm, after spending some time & tinkering around, I figured it out! Let me know so the Stack Slot Coloring should preserve LiveStack analysis?

Reply:
Yes, open a follow up PR to do that

Reply:
As there already exists a PR #93779, which deals with preserving additional analysis to be later used by RA passes, can I incorporate these changes there OR open a new follow-up PR?

Reply:
You should nearly always break it into the smallest possible separable pieces

Reply:
@vg0204 ping: are you planning to make StackSlotColoring preserve LiveStacks, or is there some reason it is hard to do?